src/index.js

```diff
@@ -48,6 +48,13 @@ function serviceFactory(swaggerConfigurator, req) {
           request.headers.correlationid = req.headers?.correlationid;
           newSpanLogger = req.gb?.logger?.loggerWithNewSpan?.();
           request.headers.span = newSpanLogger?.spanId;
+
+          const isReplaceableUA = typeof request.headers['user-agent'] === 'string'
+                                  && request.headers['user-agent'].includes('node-fetch');
+
+          if (isReplaceableUA && req.headers?.['user-agent']) {
```

Comment:
the req.headers?.['user-agent'] is unnecessary because isReplaceableUA will be false if ua is undefined or null.

Reply:
isReplaceableUA is to validate the UA in the `request` object. And Im only validating the user agent to be string since there are some UA that are not string right now.

Reply:
right, but if the header is non-existent, it will not be typeof string, and therefore false. So the second check is redundant.

Reply:
oh yes, that is to avoid replacing the requestUA with empty values, but not sure if that is a valid scenario...

Reply:
Just to clarify.. isReplaceableUA is validating `request.headers?.['user-agent']` not `req.headers?.['user-agent']`. So I added this ` && req.headers?.['user-agent']` to avoid assigning empty values to the replaceableUA.

Reply:
@djMax ☝️ are you good with this?

Reply:
Oh I see. Wow, that was some gross header naming on my part.